.github/workflows/jest-testing.yml

```diff
@@ -19,10 +19,32 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: Jest With Coverage
+      - name: Install Foundry
+        uses: foundry-rs/foundry-toolchain@v1
+        with:
+          version: nightly
+
+      - name: Install dependencies
+        run: bun install
+
+      - name: Build
+        run: bun run build
+
+      - name: Start Anvil
+        run: bun run test:anvil &
+
+      - name: Wait for Anvil
         run: |
-          bun install --frozen-lockfile
-          bun run test
+          for i in {1..30}
+          do
+            if curl -s http://localhost:8545; then
+              break
+            fi
+            sleep 1
+          done || exit 1
```

Comment:
This looks wrong. 

Normally you would do an exponential backoff. Why does it fail in the first place?

Reply:
This isn't for error handling and I've yet to see the previous step fail on that note.

Anvil being ready to use varies from instance to instance depending on multiple factors, this is just a buffer to ensure it's ready for the suite. 

30 secs is enough time for the script to try multiple instances if need be.

Also this is taken from the other test suites in use with an Anvil module.